src/app/location-selection/reef-guide-map.service.ts

```diff
@@ -67,7 +71,43 @@ export class ReefGuideMapService {
     ].filter(isDefined);
   });
 
-  constructor() { }
+  constructor() {
+    this.setupRequestInterceptor();
+  }
+
+  private setupRequestInterceptor() {
+    const apiUrl = environment.reefGuideApiUrl;
+    esriConfig.request.interceptors!.push({
+      urls: [
+        new RegExp(`^${apiUrl}`)
+      ],
+      before: params => {
+        const url = params.url as string;
+        // security double-check that we're only intercepting our API urls.
+        if (url.startsWith(apiUrl)) {
```

Comment:
this should prevent accidentally adding token to some other ArcGIS request if their `url` filter doesn't work as expected.